src/redux/reducers/backendStatus/backendStatusUpdated.js

```diff
@@ -1,13 +1,14 @@
 /* eslint-disable no-param-reassign */
-import _ from 'lodash';
 import produce from 'immer';
 
-import initialState from './initialState';
+import initialState from 'redux/reducers/backendStatus/initialState';
+
+import mergeObjectWithArrays from 'utils/mergeObjectWithArrays';
 
 const backendStatusUpdated = produce((draft, action) => {
   const { experimentId, status } = action.payload;
 
-  _.merge(draft[experimentId]?.status, status);
+  mergeObjectWithArrays(draft[experimentId]?.status, status);
```

Comment:
When we receive completedSteps array we want to replace the previous one, not merge the two arrays

Reply:
I don't understand what's happening here. I checked the `mergeObjectWithArrays` but it's not documented and the name is not clear to me. I'd appreciate it if you could elaborate and potentially add a line or two in the original `mergeObjectWithArrays` function describing what it does.

Reply:
I think if I had written the function I would call it `mergeObjectReplacingArrays` instead

Reply:
I coudl rename it

Reply:
I agree with Pol, I think it's better named something like `mergeObjectReplaceArrays`